assets/docs/configuration/targets/default.hcl

```diff
@@ -0,0 +1,24 @@
+target {
+  use "http" {
+
+    // by default no custom invalid/retryable rules so we use current default retrying strategy for target failures in `sourceWriteFunc` (in cli.go) 
```

Comment:
Is the idea that we have retries in the target, but also retain the retry that wraps it in the soruceWriteFunc?

Reply:
Yes, that was initial idea. Don't touch `sourceWriteFunc` and treat it as fallback. Then I know I don't break anything for other targets.

But now looking at other targets (e.g. for [kinesis](https://github.com/aws/aws-sdk-go/blob/main/aws/client/default_retryer.go))....clients that we use there already have retrying implemented under the hood, correct? So as we have now HTTP target covered with retries,  do we need this another layer of retrying in `sourceWriteFunc` at all and now simply make it explicit that retrying is responsibility of a target.

That would be similar to what we have in loaders.

Reply:
Well we will make sure we don't break anything for other targets regardless :D 

My initial assumption about what we'd do here was that it should replace the way we currently do retries, but I hadn't really considered what you're suggesting here.

Thinking about it now, my initial thoughts are that having a retry wrapped in a retry makes it more complicated to follow - and one of the things I'm keen for us to improve about Snowbridge is that it's quite complicated to support and there are lots of little details  like thisthat are hard to follow.

I'm not sure about the kinesis implementation but the EventHubs target is a great example of this - there's a `MaxAutoRetries` config option. This is the max retry setting for the EH client itself. But that is also wrapped in a retry - the API is very confusing here. One would reasonably expect that setting to dictate how many retries are made in total. 

But it's not, the actual max number of retries in total is 5x that setting. Why 5x? Because that's what we hardcoded into the app long before we built that target. How do we expect a user, or on of our own support team to understand this when there's some issue? Well 🤷 - we do have it in the docs I believe but good design is when the app does what someone would reasonably expect by intuition.

My gut says that there's a nice design to be found to make retries the responsibility of the target, like how you've described it. So if it's possible one outcome of this spike would be to flesh that out and get a grip on how hard it is to do (and how we can do it with confidence that we don't break things). :)